src/analysis/processing/qgsalgorithmtinmeshcreation.cpp

```diff
@@ -172,6 +176,12 @@ QVariantMap QgsTinMeshCreationAlgorithm::processAlgorithm( const QVariantMap &pa
 
   const QgsProviderMetadata *providerMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "mdal" ) );
 
+  QFileInfo fileInfo( fileName );
+  QString suffix = fileInfo.suffix();
```

Comment:
You can use QgsFileUtils::ensureFilenameHasExtension here

Reply:
done